src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1

```diff
@@ -108,53 +109,64 @@ class AgentPool: ADOSVTBase
 
     hidden [ControlResult] CheckOrgAgtAutoProvisioning([ControlResult] $controlResult)
     {
+        $controlResult.VerificationResult = [VerificationResult]::Failed
         try {
             #Only agent pools created from org setting has this settings..
-            $agentPoolsURL = "https://dev.azure.com/{0}/_apis/distributedtask/pools?poolName={1}&api-version=6.0" -f $($this.OrganizationContext.OrganizationName), $this.ResourceContext.resourcename;
-            $agentPoolsObj = [WebRequestHelper]::InvokeGetWebRequest($agentPoolsURL);
-
-            if ((($agentPoolsObj | Measure-Object).Count -gt 0) -and $agentPoolsObj.autoProvision -eq $true) {
-                $controlResult.AddMessage([VerificationResult]::Failed,"Auto-provisioning is enabled for the $($agentPoolsObj.name) agent pool.");
+            if($null -eq $this.AgentPoolOrgObj)
+            {
+                $agentPoolsURL = "https://dev.azure.com/{0}/_apis/distributedtask/pools?poolName={1}&api-version=6.0" -f $($this.OrganizationContext.OrganizationName), $this.ResourceContext.resourcename;
+                $this.AgentPoolOrgObj = [WebRequestHelper]::InvokeGetWebRequest($agentPoolsURL);
             }
-            else {
-                $controlResult.AddMessage([VerificationResult]::Passed,"Auto-provisioning is not enabled for the agent pool.");
+            
+            if($this.AgentPoolOrgObj.Count -gt 0)
+            {
+                if (($this.AgentPoolOrgObj.Count -gt 0) -and $this.AgentPoolOrgObj.autoProvision -eq $true) {
```

Comment:
as this ($this.AgentPoolOrgObj.Count -gt 0) already checked at line 121, does it needed again at 123